src/components/data-grid/DataGrid.tsx

```diff
@@ -128,22 +154,37 @@ export const DataGrid = ({
     gridApi.sizeColumnsToFit();
   };
 
-  const onSelectView = (view: DataGridView | null) => {
-    setSelectedView(view);
-    setViewState(view?.viewState!);
+  const handleActivateView = async (id: string) => {
+    const view = allViews?.find((view) => view.id === id);
+    if (view) {
+      setViewState(view?.viewState!);
+
+      if (view.id && onActivateView) {
+        await onActivateView(view.id);
+      } else if (onDeactivateView) {
```

Comment:
This shouldn't be `else if` right?

You should always deactivate the `activeView`

Reply:
I thougt backend should deactivate the old active view if you activate a new one

Reply:
Discussed with slack. Is ok.